add: like and recast reactions to `NeynarCastCard`

dylsteck opened the pull request:
this PR adds like and recast actions to the `NeynarCastCard` component by using Sign In with Neynar

the flow is as follows:
- developer sets `allowReactions` to be `true` on `NeynarCastCard`
- the user clicks the like or recast icons and the Sign In with Neynar button pops up
- upon login, the user can hit like/recast and the action will go through -- as seen by the icons changing colors(you can check on Warpcast as well)

two notes:
1) this doesn't support commenting yet but will in the future
2) with `allowReactions` set to `true`, you can also override any of the `onComment`, `onRecast`, or `onLike` functions yourself -- but if not then `onRecast` and `onLike` will default to the SIWN experience(demo'd below)

https://github.com/neynarxyz/react/assets/12853808/69cc739f-5bf2-4ccc-9fd7-4450cc9f0565

Shreyaschorge reviewed (changes requested):
@dylsteck  Minor structural changes, the rest looks good

dylsteck commented on src/components/molecules/CastCard/index.tsx:
```diff
@@ -0,0 +1,196 @@
+  color: theme.vars.palette.textMuted,
+}));
+
+const Main = styled.div(() => ({
```
for sure -- i still might need to wrap it or add some extra styling though because the space-between logic for this `Main` const separates these icons below
<img width="580" alt="image" src="https://github.com/neynarxyz/react/assets/12853808/e0514c01-de92-4e52-93e6-28a8f20f062c">

dylsteck commented:
@Shreyaschorge i made most of the changes from the comments you left(made one small comment) and will work on following the [atomic design structures](https://atomicdesign.bradfrost.com/chapter-2/) more going forward. i just need to make one more change on the auth popover on `Reactions.tsx` and then this PR should be good to be reviewed again.

dylsteck commented:
just updated the PR to address the comments & also changed the styling of the auth `Popover` (below)

<img width="688" alt="Screenshot 2024-07-05 at 10 45 23 AM" src="https://github.com/neynarxyz/react/assets/12853808/2b0b3395-8e6c-46a7-88e9-8b611373c183">

Shreyaschorge commented on src/components/atoms/Reactions/index.tsx:
```diff
@@ -0,0 +1,203 @@
+  onLike?: () => void;
+};
+
+const Reactions: React.FC<ReactionsProps> = ({
```
@dylsteck 
No need to change this now, we can ship it with it.
Reactions should be a molecule-level component (idea being multiple atoms --reactions in this case-- form a molecule)

dylsteck replied:
ok gotcha, for the next round of edits in the future I'll make `Reactions` a molecule and the individual reactions themselves atoms
